### ISSUE TITLE
Multiple Actions in the same query are synchronous rather than parallel 

### ISSUE THREAD

**Issue by maitham** [hasura/graphql-engine#5023]
Hey, 

We've assumed multiple queries in the same graphql statement would happen in parallel as per the docs. We get this behaviour with normal queries, however if the queries contain an action it seems these happen sequentially not in parallel. Is this expected behaviour?

**Comment by tirumaraiselvan** [comment]
> We've assumed multiple queries in the same graphql statement would happen in parallel as per the docs.

Can you point us to the docs?

**Comment by maitham** [comment]
https://hasura.io/docs/1.0/graphql/manual/queries/multiple-queries.html

**Comment by tirumaraiselvan** [comment]
This was actually a mistake in the documentation and is being fixed here: https://github.com/hasura/graphql-engine/pull/5105

Normal queries and actions, both are run sequentially.

**Comment by maitham** [comment]
Ah thats a shame. Are there any plans to support parallel queries? It seems to be a given that this should be the behaviour a tleast for actions

**Comment by tirumaraiselvan** [comment]
Yes, I think we should be able to do that in the future. Will change this issue into a feature request.

**Comment by rccoe** [comment]
Not 100% the same feature request, but very much in the same vein: I want to run an action at the end of a CRUD operation that depends on the results of the prior operation. 

It needs to be synchronous, so can't run after. The alternatives are
* Bring the CRUD logic into an action -- It seems like overkill to have to create 4 actions with a bunch of boilerplate webhook code
* Chain the requests client-side

It would be great to somehow tell Hasura to close the DB transaction before triggering the action.

**Comment by harpyng** [comment]
Any update on this request?

**Comment by manasag** [comment]
Hi, I want to update that we are launching Hasura V3 (DDN) this month, and it would be worth noting that parallel and batch execution is something that has been built in a great way in V3 (both queries and actions). The query tracing feature also provides a good view of this parallelisation. In V3, actions are now supported as custom Typescript code driven modules that allows for a lot of customisation around business logic and runtime parallelisation (using NodeJS executor). 

On that note, we're celebrating our DDN beta launch at Hasura Dev Day on April 16 :tada:  Join us to see the full power of Hasura DDN. Sign up here: https://hasura.io/dev-day. Can't make it? Sign up anyway and we'll send you the recording.
